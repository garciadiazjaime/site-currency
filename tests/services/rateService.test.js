import {
  getRates
} from '../../services/rateService'

describe('rateService', () => {
  describe('getRates', () => {
    beforeEach(() => {
      global.fetch = jest.fn(() => ({
        json() {
          return 'happy-rate'
        }
      }))
    })

    afterEach(() => {
      global.fetch.mockRestore()
    })

    it('returns expected values', async () => {
      const response = await getRates()

      expect(global.fetch.mock.calls).toMatchSnapshot()
      expect(response).toEqual({
        'rates': 'happy-rate',
        'usdmxn': 'happy-rate'
      })
    })
  })
})
