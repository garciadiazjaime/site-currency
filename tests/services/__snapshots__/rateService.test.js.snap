// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rateService getRates returns expected values 1`] = `
Array [
  Array [
    "http://127.0.0.1:3000/rates",
  ],
  Array [
    "http://127.0.0.1:3000/rates/usdmxn",
  ],
]
`;
